code/edge/src/main/java/com/adobe/marketing/mobile/EdgeExtension.java

```diff
@@ -182,9 +184,19 @@ public boolean readyForEvent(@NonNull Event event) {
 	 * Valid Configuration and Identity shared states are required for processing the event. If a valid Configuration shared state is
 	 * available, but no {@code edge.configId } is found or {@link #shouldIgnore(Event)}` returns true, the event is dropped.
 	 *
-	 * @param event an event containing {@link ExperienceEvent} data for processing; the event and the event data should not be null, checking in listener
+	 * @param event an event containing {@link ExperienceEvent} data for processing; the event data should not be null/empty
 	 */
 	void handleExperienceEventRequest(@NonNull final Event event) {
+		if (Utils.isNullOrEmpty(event.getEventData())) {
```

Comment:
These checks were moved from listeners to handlers, I compared them with the iOS implementation also and they now match that. The only handler that does not check for event data payload is getLocationHint, seems like we don't do that check in iOS.